src/argilla/training/peft.py

```diff
@@ -0,0 +1,256 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from typing import List, Union
+
+import numpy as np
+
+import argilla as rg
+from argilla.training.transformers import ArgillaTransformersTrainer
+from argilla.training.utils import (
+    filter_allowed_args,
+)
+from argilla.utils.dependency import require_version
+
+
+class ArgillaTransformersPEFTTrainer(ArgillaTransformersTrainer):
+    _logger = logging.getLogger("ArgillaTransformersPEFTTrainer")
+    _logger.setLevel(logging.INFO)
+
+    import sys
+
+    if sys.version_info[1] < 9 or sys.version_info[0] < 3:
+        raise Exception("Must be using Python 3.9 or higher or PEFT won't work.")
+
+    require_version("peft")
+
+    def init_training_args(self):
+        super().init_training_args()
+
+        self.lora_kwargs = {
+            "r": 8,
+            "target_modules": None,
+            "lora_alpha": 16,
+            "lora_dropout": 0.1,
+            "fan_in_fan_out": False,
+            "bias": "none",
+            "inference_mode": False,
+            "modules_to_save": None,
+            "init_lora_weights": True,
+        }
+
+    def init_model(self, new: bool = False):
+        from peft import LoraConfig, PeftConfig, PeftModel, get_peft_model
+        from transformers import AutoTokenizer
+
+        if self._record_class == rg.TextClassificationRecord:
+            self.lora_kwargs["task_type"] = "SEQ_CLS"
+        elif self._record_class == rg.TokenClassificationRecord:
+            self.lora_kwargs["task_type"] = "TOKEN_CLS"
+        else:
+            raise NotImplementedError("This is not implemented yet.")
+
+        try:
+            config = PeftConfig.from_pretrained(self.model_kwargs["pretrained_model_name_or_path"])
+            model = self._model_class.from_pretrained(
+                config.base_model_name_or_path,
+                return_dict=True,
+                num_labels=len(self._id2label),
+                id2label=self._id2label,
+                label2id=self._label2id,
+            )
+            self._model_sub_class = model.__class__
+            model = PeftModel.from_pretrained(model, self.model_kwargs["pretrained_model_name_or_path"])
+        except Exception:
```

Comment:
Since we're not catching any exception I'd say we're good with

```suggestion
        except:
```